### PR TITLE
TF-1360 Fix can not load new email until reload mail list

### PR DESCRIPTION
**Pull request by dab246**
#1360 

- [x] Fix can not load new email until reload mail list
- [x] Fix open again email detailed when logout app opened from notification
- [x] Fix can not MarkAsRead when open app from notification
- [x] Fix can not add data to stream controller when re-login

### Resolved

- On Foreground & Background

https://user-images.githubusercontent.com/80730648/211258697-a46cf8ca-1f78-47c0-9133-05e5092f2ffc.mp4

- On Terminated

https://user-images.githubusercontent.com/80730648/211258664-d0ab5b82-d32c-4877-825a-937d4a89bf45.mov

### CODE REVIEW

**Review by nqhhdev — APPROVED** [review]

**Review by hoangdat — CHANGES_REQUESTED** [review]
still waiting for better solution in handling empty stream

**Comment by hoangdat** [comment]
@dab246 can you write the docs related to `streamController` in re-login case?

**Comment by dab246** [comment]
> @dab246 can you write the docs related to `streamController` in re-login case?

It's pretty simple, so I'll explain it here:
- When we do `logout`, call the function `closeStreamController` of fcm. So when `login` again we have not initialized `StreamController` because these objects are initialized in the `singleton` class.
- So the solution would be to re-initialize the `StreamController` when it is `closed` every time I login again

**Review by sherlockvn — DISMISSED** [review]

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart [hunk 1]
```diff
@@ -1371,36 +1382,62 @@ class MailboxDashBoardController extends ReloadableController {
...
+
+  void handleOnFocusGained() {
+    log('MailboxDashBoardController::handleOnFocusGained():');
+    if (!BuildUtils.isWeb) {
+      _updateTheme();
+    }
+    refreshActionWhenBackToApp();
+  }
+
+  void _updateTheme() {
```
[the remark this answers is not in the document]
Comment (dab246) [reply lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:1450]: On android, `statusbar`, `bottombar` difference between screens. So we need change it when resume app from background

**Review by hoangdat — APPROVED** [review]

**Pull request merged**